### PR TITLE
ExpireSnapshots: do not commit when no snapshots are expired.

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

**Comment by rdblue** [comment]
@danielcweeks, can you review this?

File: core/src/main/java/com/netflix/iceberg/RemoveSnapshots.java [hunk 1]
```diff
@@ -115,7 +115,10 @@ public void commit() {
         .onlyRetryOn(CommitFailedException.class)
         .run(item -> {
           TableMetadata updated = internalApply();
-          ops.commit(base, updated);
+          // only commit the updated metadata if at least one snapshot was removed
+          if (updated.snapshots().size() != base.snapshots().size()) {
```
Comment (rdsr) [inline core/src/main/java/com/netflix/iceberg/RemoveSnapshots.java]: Would it be more precise to check for `if (updated.snapshots().size() < base.snapshots().size())`?

Comment (danielcweeks) [reply core/src/main/java/com/netflix/iceberg/RemoveSnapshots.java]: I think those two checks are functionally equivalent for this check.  This is to effectively check that the number of snapshots has not changed.  I would err on the side of equality as the proposed change would conceptually mean that you could increase the number of snapshots, which would not be expected in this case.

**Comment by danielcweeks** [comment]
+1

**Comment by rdblue** [comment]
Thanks for the reviews!

**Pull request merged**